scripts/scr_battle_roster/scr_battle_roster.gml

```diff
@@ -42,32 +42,40 @@ function scr_battle_roster(_battle_loci, _loci_specific, _is_planet) {
 
 
 	// if (_is_planet=true){
-	    var company, v, meat;
+	    var company, v, meat, marine;
 	    company=-1;v=0;meat=false;
+		deploying_unit=obj_ini;
 
 	    instance_activate_object(obj_pnunit);
 
+		//For each company and the HQ
 	    repeat(11){
 			if (man_limit_reached == true){break;}
 	        company+=1;v=0;okay=0;
-	        repeat(300){
+			//For each marine in that company, while unit exists (either marine name or vehicle role, vehicles have no names saved)
+			//Marines and vehicles get added AT THE SAME TIME, (index [0][1] adds marine AND vehicle at index at the same time for loop x)
+			//This is possible since array for saving vehicles and marines are separated
+			//v<300 based on previous repeat(300), so it might be nonsense
+			while ((deploying_unit.name[company,v]!=""      || 
+					deploying_unit.veh_role[company,v]!="")    && v<300){
 				if (man_limit_reached == true){break;}
+				//array[0] set to 0, so the proper array starts at array[1], for some reason
 				v+=1;
 
 	            if (stop=0){
 	                if (!instance_exists(obj_drop_select)){// Only when attacked
-	                    if (_is_planet=true) and (obj_ini.loc[company,v]=fiy) and (obj_ini.wid[company,v]=secc) and (obj_ini.hp[company,v]>0) and (obj_ini.god[company,v]<10) then okay=1;
-	                    if (_is_planet=false) and (obj_ini.lid[company,v]=_loci_specific) and (_loci_specific=_loci_specific) and (obj_ini.hp[company,v]>0) and (obj_ini.god[company,v]<10) then okay=1;
+	                    if (_is_planet=true) and (deploying_unit.loc[company,v]=fiy) and (deploying_unit.wid[company,v]=secc) and (deploying_unit.hp[company,v]>0) and (deploying_unit.god[company,v]<10) then okay=1;
```

Comment:
can we just use: if(_is_planet) ? or GML requires code to be explicit about comparisons?

Reply:
We could (IDE does not complain at least). I am somewhat reticent to change it, since I find it more readable like this and as far as I can remember, Boolean checks are explicit on other parts of the code.

Reply:
IMO as long as the variable starts with a question, like _is_planet, or _has_enough_marines, then it's very readable and actually looks almost like a natural sentence, i.e. 

if(_player_has_enough_requisition) 

looks better than 

if(_player_has_enough_requisition == true)

Reply:
Refactored it.